packages/babel-plugin-minify-simplify/src/index.js

```diff
@@ -28,6 +28,62 @@ module.exports = ({ types: t }) => {
   const or = (a, b) => t.logicalExpression("||", a, b);
   const and = (a, b) => t.logicalExpression("&&", a, b);
 
+  const operators = new Set([
+    '+', '-', '*', '%',
+    '<<', '>>', '>>>',
+    '&', '|', '^', '/',
+    '**'
+  ]);
+
+  const updateOperators = new Set([
+    '+', '-'
+  ]);
+
+  function isEqual(arr1, arr2) {
```

Comment:
Ugh, good catch. So the reason it works is because property names are resolved RTL so `foo.bar` won't match `foo.bar.baz` due to arrays being `['bar', 'foo']` and `['baz', 'bar', 'foo']`. We can maybe do an additional length check which could be a nice compromise between speed and correctness?